### ISSUE TITLE
Tool crashes on link object with empty JSON object

### ISSUE THREAD

**Issue by jautor** [DMTF/Redfish-Service-Validator#319]
Tool crashes on an unhandled exception for a 'split' for a URI decode if a link object has no properties.  While that is an error (not valid Redfish format), the case should be handled.

Example
```
   "Thermal": {},
```
where correct form should be:
```
   "Thermal": {
       "@odata.id": "/redfish/v1/Chassis/VoteQuimby/Thermal"
   },
```

**Comment by billdodd** [comment]
@jautor Can you provide some log output showing the crash exception callstack?

I can't reproduce this. With the example you show, I get an error like this:

```
ERROR - Thermal: EntityType resource does not contain required @odata.id property, attempting default /redfish/v1/Chassis/1U/Thermal
```

But no crash.

**Comment by jautor** [comment]
Yes, I'll reproduce and capture the call stack.  Nothing in the logfile...

**Comment by jautor** [comment]
```
Traceback (most recent call last):
  File "RedfishServiceValidator.py", line 1442, in <module>
    status_code, lastResultsPage, exit_string = main()
  File "RedfishServiceValidator.py", line 1390, in main
    success, counts, results, xlinks, topobj = validateURITree('/redfish/v1/', '
ServiceRoot', expectedJson=jsonData)
  File "RedfishServiceValidator.py", line 1213, in validateURITree
    success, linkCounts, linkResults, xlinks, xobj = executeLink(links[linkName]
, thisobj)
  File "RedfishServiceValidator.py", line 1186, in executeLink
    linkURI, uriName + ' -> ' + linkName, parent=parent, allLinks=allLinks)
  File "RedfishServiceValidator.py", line 1213, in validateURITree
    success, linkCounts, linkResults, xlinks, xobj = executeLink(links[linkName]
, thisobj)
  File "RedfishServiceValidator.py", line 1186, in executeLink
    linkURI, uriName + ' -> ' + linkName, parent=parent, allLinks=allLinks)
  File "RedfishServiceValidator.py", line 1213, in validateURITree
    success, linkCounts, linkResults, xlinks, xobj = executeLink(links[linkName]
, thisobj)
  File "RedfishServiceValidator.py", line 1186, in executeLink
    linkURI, uriName + ' -> ' + linkName, parent=parent, allLinks=allLinks)
  File "RedfishServiceValidator.py", line 1213, in validateURITree
    success, linkCounts, linkResults, xlinks, xobj = executeLink(links[linkName]
, thisobj)
  File "RedfishServiceValidator.py", line 1186, in executeLink
    linkURI, uriName + ' -> ' + linkName, parent=parent, allLinks=allLinks)
  File "RedfishServiceValidator.py", line 1202, in validateURITree
    elif links[linkName].uri.split('#')[0].endswith('/'):
AttributeError: 'NoneType' object has no attribute 'split'
```
@billdodd this type of exception seems to be a common Python trap as we're parsing URIs and splitting them in many places.  We should check that we have handlers around these operations as malformed / missing / invalid URIs is certainly a common Redfish implementation issue.  So we'd like the tool to survive those bugs to report them!

**Comment by jautor** [comment]
More details - this particular error seems to occur when the tool fills in a missing/malformed URI for the odata.id - but in this case the resource is actually not present either (the tool assumption is a good one as it allows testing to continue in a lot of cases).  So a call to that assumed URI will return a 404, and it's this cases that appears to produce the 'NoneType' object that fails on the .spilt()

**Comment by billdodd** [comment]
Thanks, Jeff. This was reported and fixed last month (issue #309, PR #310). That's why I wasn't able to reproduce. Can you try again with the current version?

**Comment by jautor** [comment]
D'oh!  Thought I was always up to date - but I'm staring at the wrong Github repo...  Now running with 1.2.7 and confirm that it's been fixed.  Closing no fix needed.
